packages/core/src/config/parse.ts

```diff
@@ -2590,7 +2618,44 @@ const assertAvailableCreate3Addresses = async (
       contractConfig.kind === 'no-proxy' &&
       (await provider.getCode(contractConfig.address)) !== '0x'
     ) {
-      unavailable.push(referenceName)
+      const { bytecode, abi } = readContractArtifact(
+        artifactPaths[referenceName].contractArtifactPath,
+        integration
+      )
+
+      const deployedHash = await getDeployedCreationCodeWithArgsHash(
+        provider,
+        parsedConfig.options.organizationID,
+        referenceName,
+        contractConfig.address
+      )
+
+      if (!deployedHash) {
+        throw new Error(
+          `Could not retrieve creation code hash for deployed contract. Should never happen.`
+        )
+      }
+
+      const currHash = ethers.utils.keccak256(
+        getCreationCodeWithConstructorArgs(
+          bytecode,
+          contractConfig.constructorArgs,
+          abi
+        )
+      )
+
+      const match = BigNumber.from(deployedHash).eq(BigNumber.from(currHash))
+      if (match) {
+        logValidationError(
+          'warning',
+          `Skipping deployment of ${referenceName}. Add a new 'salt' value to re-deploy it at a new address.`,
```

Comment:
Nit: Would be nice to include a little bit more context on exactly why it was skipped. 

```suggestion
          `Skipping deployment of ${referenceName} since it has already been deployed and has not changed. Add a new 'salt' value to re-deploy it at a new address.`,
```